### DISCUSSION TITLE
Constrained Proof of Stake

### DISCUSSION THREAD

**Discussion by F041** [deso-protocol/dips#8] [Medium CIPs]
---
cip: <to be assigned>
title: Constrained Proof of Stake
author:  Gabriele GIordano <@F041>, @gabrielegio on Bitclout
discussions-url: <8>
created: 2021-07-09
---

## One-Line Description
A mathematical function so that PoS **won't** exceed 100k$ per year. It drops the interest rates very quickly after a certain amount of staked clout. The starting interest rate results relatively high at the minimum amount of staked clout to get the benefits, but not particularly dissimilar to services that give APY like BlockFi.

## Explanation and Motivation
We know economical inequalities have reached a local maxima high since the 1900s. In some countries they will even surpass the historical record soon. 
The crypto economy exacerbates those inequalities.
PoW has high entrance barriers. 

S shaped functions, paired with PoS rewards, can create an agreed mechanism for the distribution of the spent nanos, and sustain a local equality. 

## Specification
[S shaped functions](https://en.wikipedia.org/wiki/Generalised_logistic_function) for staking rewards, with upper asymptote based on the function of [wellbeing-income](https://ourworldindata.org/happiness-and-life-satisfaction#the-link-within-countries), and lower asymptote, 100$ USD in CLOUT. 

![image](https://user-images.githubusercontent.com/19621180/125074178-3a827100-e0bd-11eb-8bd3-695d151746a9.png)

Which CLOUT to get considered staked? Just one's creator coin or every position in one's wallet? Up for discussion. 

Use this spreadsheet to play with data, I suggest the generalized logistic sheet:
https://docs.google.com/spreadsheets/d/1_myFhiZkfJeRCiRaCHOGgI6idgM0yAHQVbFBVhpTHi0/edit?usp=drivesdk

For other details, check: https://bitclout.com/posts/33bfd2cbcc66778b08639a8df755c7f3202cfccf164f17550d10510ff5d54c9f

I have no idea on how to implement it in the core, since I just played with the mathematical concept behind the idea.

## Backwards Compatibility
No idea 

## Test Cases
John has 1k$ staked in clout. He would get an APR, distributed monthly, between 12,7% and 12,5%, using the formula indicated in the spreadsheet mentioned previously. 

## Security Considerations
To avoid rent seeking capitalism, the [bottleneck](https://bitclout.com/posts/68e7d84c23b97f7de0c19752829d95de221d38d33ae9c23b3377e7c805405e5b) problem between rewards and spent nanos, and not only, one can use the PoS+ protocol to ensure just active users (with an agreed metrics, otherwise bots can automate that) get the rewards. From page 80 of the Symbol [white paper](https://nemtech.github.io/catapult-whitepaper/main.pdf) you can see an idea for the weighting algorithm for this purpose. 
This would also solve the problem of big accounts splitting their CLOUT across n accounts to reap higher interests. Imagine for example an 1M$ account spreading clout on 1000 accounts so they get more yearly interests. 
EDIT: PoS+ paired with GoodDollar videoselfie once in a while could solve the mentioned problems. Videoselfies would keep put from staking non humans accounts, like projects, wallet profiles, bots, etc. 
https://www.gooddollar.org/digital-photo-authentication-face-verification/

**Comment by HPaulson** [comment]
I think this post needs to be much much more detailed, outlining a clear problem, solution, specifics of such. and other considerations. It's not too clear what you're proposing specifically, and the motivation behind it.

**Reply by bloated-devish** [reply]
+1 to everything HPaulson said. Additionally - in the format that maebeam has requested. 

This definitely belongs to the large bucket.

**Reply by dgsus** [reply]
agree, this is a huge theme that needs to be discussed extensively and with great detail.

very important thing for that agenda is indeed - finding a  PoS consensus mechanism that doesnt doesnt only make the rich richer and also taking in account this novel social blockchain.

**Reply by F041** [reply]
Updated. Waiting your next feedbacks

**Reply by HPaulson** [reply]
@F041 Couple concerns:

1. Incentivizing. By doing some quick math, it's clear that staking more would be less favorable to staking less. For example, by staking $50,000 you'd get a $1,230 return (2.46% yield), and by staking $75,000 you'd get a $397.5 return (.53% yield). At .01% for $100,000 stake, you'd make $10. Can you start to see where this makes little sense? While the economics here work out up to the $50,000 range, the don't following. This means there's a sweet spot of where the most gain, not yield, could be achieved -- as staking more on this curve gives you less actual return somewhere between the $50k and $75k mark. This would cause less overall clout to be staked, and especially cause problems for staking pools - problems I don't think we want.

2. My other concern here is scale. How does a bottleneck of $100k serve us at a scale, say for example BitClout grows to the size of Twitter, the price would grow exponentially -- so would something like $100k be a large enough cap?

**Reply by F041** [reply]
1. > staking more would be less favorable to staking less. 

no
Please check the spreadsheet. Or look again at the graph.

2. One receives maximum 100k$ in Clout. Nothing forbids them to keep them so they can go up in price and sell them for more USD. If I understood your concern.

**Reply by HPaulson** [reply]
My apologies -- I read this incorrectly. The way I read this was the max STAKE was $100k, not max reward. Lemme read this over again

**Reply by HPaulson** [reply]
I see okay. After reading over this once more & checking out your doc (Before just looked at the graph) it makes much much more sense -- and seems favorable. Two concerns:

1. In concerns you mention stopping the ability to spread clout across multiple accounts based on activity metrics. What are some ways we could measure this? Additionally, is this a negative aspect, as some accounts such as VCs or regular investors who don't wish to use the platform may remain inactive? Should they not get staking rights? In my view -- they most definitely should have the ability to stake, and we should incentivize them to do so.

2. Does it makes sense to include Creator Coins here? Since CCs are essentially clout which is already "staked" (For better or worse) in a user, I don't think it makes much sense to also have them useable in PoS. It makes much more sense, from my viewpoint, to only allow staking of pure Clout.

**Reply by F041** [reply]
1. Number of transactions (messages, posts, likes, diamonds). So more options than the formula mentioned in the Symbol whitepaper, but leaves open the problem for bots. I find weird to have a social and not use it. Then better to leave the clout in exchanges, for the cost of reduced security.
2. staked CCs can signal trust in a creator. If you remeber, weeks ago, someone talked about locking/disclousing the time period of investments. If you stake pure Clout instead of CCs, you waste more capital, using the words of maxis (and I agree with them).

**Reply by bloated-devish** [reply]
I am in agreement with @HPaulson 

> 1. Should they not get staking rights? In my

@F041 : It sounds like we are trying to solve societal problems (like social inequality) with BC. Not that we shouldn't, but that's not the primary goal of this project.
Are we encouraging large VCs to quit the platform, so they can take the money and stake elsewhere (because the gains elsewhere would be better - in steady state)? If anything VCs are often (but not always) a large time horizon investors and are diamond hands, unlike paper hands that regulars like me are. So I am a little confused what this constrained PoS achieves?

**Reply by F041** [reply]
A constrained PoS won't surely solve societal problems, it could just show an angle of the internet with more equality. 

VCs have anyway more profitable ways to make their capital work without staking. I don't agree that this proposal will make them flee, since there exists a minor trend in sustainable (in many ways) investments. One core example: the Great Reset. 

Constrained PoS incentives users to use their capital here instead of somewhere else, especially if relatively low, creating a much less unequal population compared to other crypto environments. As a side effect it provides free PR. It winks to the Keynesians out there, and those libertarians that still think that constraints in a free market reduce distortions and increase stability (I forgot the labels).

**Reply by F041** [reply]
Updated in the last part
